openedx/features/course_experience/templates/course_experience/course-outline-fragment.html

```diff
@@ -38,14 +38,80 @@
                                 href="${ subsection['lms_web_url'] }"
                                 id="${ subsection['id'] }"
                             >
-                                <span>${ subsection['display_name'] }</span>
-                                <span class="sr-only">${ _("This is your last visited course section.") }</span>
-                                % if subsection['current']:
-                                    <span class="resume-right">
-                                        <b>${ _("Resume Course") }</b>
-                                        <span class="icon fa fa-arrow-circle-right" aria-hidden="true"></span>
-                                    </span>
-                                % endif
+                                <div class="subsection-text">
+                                    ## Subsection title
+                                    <span class="subsection-title">${ subsection['display_name'] }</span>
+
+                                    <div class="details">
+                                        ## There are behavior differences between rendering of subsections which have
+                                        ## exams (timed, graded, etc) and those that do not.
+                                        ##
+                                        ## Exam subsections expose exam status message field as well as a status icon
+                                        <%
+                                            if subsection.get('due') is None:
+                                                data_string = subsection.get('format')
+                                            else:
+                                                if 'special_exam_info' in subsection:
+                                                    data_string = _('due {date}')
+                                                else:
+                                                    data_string = _("{subsection_format} due {{date}}").format(subsection_format=subsection.get('format'))
+                                        %>
+                                        % if subsection.get('format') or 'special_exam_info' in subsection:
```

Comment:
@bjacobel Do we need an empty `<div class="details">` if there are no details, or is it ok to use this if to gate the entire div?